### ISSUE TITLE
Doesn't handle assets from gems (jekyll themes and plugins)

### ISSUE THREAD

**Issue by sim642** [RobertoJBeltran/jekyll-terser#3]
As the README says, this is based on https://github.com/zroger/jekyll-less, which has been unmaintained for 10 years. Crucially, its architecture seems to be such that it doesn't correctly handle assets from gems (jekyll themes and plugins).

Whereas https://github.com/digitalsparky/jekyll-minifier seems to use different architecture that handles gem assets correctly. It might be useful to borrow the internals from there that don't make 10-year-old assumptions about how jekyll's static files seems to work.
Although https://github.com/digitalsparky/jekyll-minifier/pull/62 might make the whole effort redundant as well.

**Comment by RobertoJBeltran** [comment]
Is there any reason you would personally prefer to use this project over the other one you linked?

**Comment by sim642** [comment]
No, but https://github.com/alshedivat/al-folio switched to this. Personally I'm hoping for https://github.com/digitalsparky/jekyll-minifier/pull/62, but until then this issue at least documents the limitation, even if it doesn't end up being fixed.

**Comment by RobertoJBeltran** [comment]
So you're a user of al-folio? Was this a public discussion? If so, could you link me to it?

**Comment by george-gca** [comment]
Yes, we had discussed about switching to `jekyll-terser` or keep using `jekyll-minifier`. All discussion happened in https://github.com/alshedivat/al-folio/pull/2571, and I even made a PR for both projects, as commented in https://github.com/alshedivat/al-folio/pull/2571#issuecomment-2227154029. Also we kept using an internal copy of this plugin in `al-folio` as I was waiting for a rubygem to be available, but we are switching to use the git repo in https://github.com/alshedivat/al-folio/pull/2945.

**Comment by RobertoJBeltran** [comment]
Thank you @george-gca I'll be sure to review this. 

@sim642 Let me know what specifications would constitute correct handling of assets from gems (jekyll themes and plugins) so I can properly assess this issue. Thanks for your help.
